### PR TITLE
Multiple primaries

### PR DESCRIPTION
**Pull request by eggrobin**
This greatly improves the equipotentials for the outer planets, as illustrated by screenshots for Neptune below.

The computation is much slower for Neptune than for Earth, and is fastest for the Moon (since we need to evaluate 31, 5, and 2 trajectories respectively). We will want to look at optimizing this somehow.
| Before | After |
|---|---|
| screenshot8 | screenshot9 |

### CODE REVIEW

File: physics/barycentric_rotating_reference_frame_body.hpp [hunk 1]
```diff
@@ -33,29 +33,40 @@ BarycentricRotatingReferenceFrame<InertialFrame, ThisFrame>::
...
-        not_null<MassiveBody const*> primary,
+        std::vector<not_null<MassiveBody const*>> primaries,
         std::vector<not_null<MassiveBody const*>> secondaries)
     : ephemeris_(std::move(ephemeris)),
-      primary_(std::move(primary)),
+      primaries_(std::move(primaries)),
       secondaries_(std::move(secondaries)),
-      primary_trajectory_(ephemeris_->trajectory(primary_)) {
+      primary_trajectory_(ephemeris_->trajectory(primaries_.front())) {
+  std::set<not_null<MassiveBody const*>> primary_set(primaries_.begin(),
```
Comment (pleroy) [inline physics/barycentric_rotating_reference_frame_body.hpp:49]: `btree_set` (3x).

File: physics/rotating_pulsating_reference_frame_body.hpp [hunk 2]
```diff
@@ -23,29 +23,40 @@ RotatingPulsatingReferenceFrame<InertialFrame, ThisFrame>::
...
-        not_null<MassiveBody const*> const primary,
+        std::vector<not_null<MassiveBody const*>> primaries,
         std::vector<not_null<MassiveBody const*>> secondaries)
     : ephemeris_(ephemeris),
-      primary_(primary),
+      primaries_(std::move(primaries)),
       secondaries_(std::move(secondaries)),
-      rotating_frame_(ephemeris_, primary_, secondaries_) {
+      rotating_frame_(ephemeris_, primaries_, secondaries_) {
+  std::set<not_null<MassiveBody const*>> primary_set(primaries_.begin(),
```
Comment (pleroy) [inline physics/rotating_pulsating_reference_frame_body.hpp:39]: `btree_set`

File: physics/rotating_pulsating_reference_frame_body.hpp [hunk 3]
```diff
@@ -172,9 +189,15 @@ RotatingPulsatingReferenceFrame<InertialFrame, ThisFrame>::r_derivatives(
         ephemeris_->trajectory(secondary)->EvaluatePosition(t),
         secondary->gravitational_parameter());
   }
+  BarycentreCalculator<Position<InertialFrame>, GravitationalParameter>
+      primary_position;
```
Comment (pleroy) [inline physics/rotating_pulsating_reference_frame_body.hpp:193]: Would fit on the previous line?  (Many places in this PR.)

File: ksp_plugin/plugin.cpp [hunk 4]
```diff
@@ -1184,17 +1184,21 @@ Plugin::NewBodySurfaceNavigationFrame(
 
 not_null<std::unique_ptr<PlottingFrame>>
 Plugin::NewRotatingPulsatingPlottingFrame(
-    Index const primary_index,
-    std::vector<Index> const secondary_indices) const {
-  Celestial const& primary = *FindOrDie(celestials_, primary_index);
+    std::vector<Index> const& primary_indices,
+    std::vector<Index> const& secondary_indices) const {
+  std::vector<not_null<MassiveBody const*>> primaries;
+  for (Index const& i : primary_indices) {
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp:1190]: Why `const &`?

File: physics/rotating_pulsating_reference_frame_body.hpp (hunk 3 above)
Comment (eggrobin) [reply physics/rotating_pulsating_reference_frame_body.hpp:193]: No.

**Pull request merged**